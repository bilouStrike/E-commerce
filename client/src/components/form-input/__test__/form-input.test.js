import { shallow, mount } from 'enzyme';
import React from 'react';
import toJson from 'enzyme-to-json';
import FormInput from '../form-input.component';

describe('FormInput compoent', () => {
    const MockhandleChange = jest.fn();
    let mockLabel;
    let wrapper;
    let mockOtherProps = {
        name: 'email',
        type: 'email',
        value: 'value',
        required: true,
        label: 'email'
    }

    it('Should render FormInput without label', () => {
        mockLabel = false;
        wrapper = toJson(shallow(<FormInput handleChange={MockhandleChange} label={mockLabel} {...mockOtherProps}  />))
        expect(wrapper).toMatchSnapshot();
    });

    it('Should render FormInput with label', () => {
        mockLabel = true;
        wrapper = toJson(shallow(<FormInput handleChange={MockhandleChange} label={mockLabel} {...mockOtherProps}  />))
        expect(wrapper).toMatchSnapshot();
    });

    it('Should render FormInput with label and without shink class', () => {
        mockLabel = true;
        mockOtherProps.value = 0;
        wrapper = toJson(shallow(<FormInput handleChange={MockhandleChange} label={mockLabel} {...mockOtherProps}  />))
        expect(wrapper).toMatchSnapshot();
    });

    it('expect submit onclick', () => {
        wrapper = shallow(<FormInput handleChange={MockhandleChange} label={mockLabel} {...mockOtherProps}  />);
        wrapper.find('input[type="email"]').simulate('change', {
            target: {
              value: 'newvalue',
            },
        });
        expect(MockhandleChange).toBeCalled();
    });
});