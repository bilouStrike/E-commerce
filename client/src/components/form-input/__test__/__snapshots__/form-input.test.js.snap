// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormInput compoent Should render FormInput with label 1`] = `
<div
  className="group"
>
  <input
    className="form-input"
    name="email"
    onChange={[MockFunction]}
    required={true}
    type="email"
    value="value"
  />
  <label
    className="shrink  form-input-label"
  >
    email
  </label>
</div>
`;

exports[`FormInput compoent Should render FormInput with label and without shink class 1`] = `
<div
  className="group"
>
  <input
    className="form-input"
    name="email"
    onChange={[MockFunction]}
    required={true}
    type="email"
    value={0}
  />
  <label
    className="  form-input-label"
  >
    email
  </label>
</div>
`;

exports[`FormInput compoent Should render FormInput without label 1`] = `
<div
  className="group"
>
  <input
    className="form-input"
    name="email"
    onChange={[MockFunction]}
    required={true}
    type="email"
    value="value"
  />
  <label
    className="shrink  form-input-label"
  >
    email
  </label>
</div>
`;
